examples/plot_linear_regression.py

```diff
@@ -0,0 +1,151 @@
+# -- import necessary packages and modules
+from collections import namedtuple
+
+import numpy as np
+from scipy import linalg, stats
+from sklearn.linear_model import LinearRegression
+
+from mne.datasets import limo
+from mne.stats import linear_regression
+# from mne.source_estimate import SourceEstimate
+from mne.evoked import EvokedArray
+from mne.io.pick import pick_types
+
+###############################################################################
+# list with subjects ids that should be imported
+subjects = list(range(2, 3))
+# create a dictionary containing participants data
+limo_epochs = {str(subj): limo.load_data(subject=subj) for subj in subjects}
+
+# get key from subjects dict for easy slicing
+# subjects = list(limo_epochs.keys())
+
+# interpolate missing channels
+for subject in limo_epochs.values():
+    subject.interpolate_bads(reset_bads=True)
+
+# pick channels that should be included in the analysis
+picks_eeg = pick_types(limo_epochs['2'].info, eeg=True)
+# channels to be excluded
+exclude = ['EXG1', 'EXG2', 'EXG3', 'EXG4']
+
+# save epochs information (needed for creating a homologous
+# epochs object containing linear regression result)
+epochs_info = limo_epochs['2'].copy().drop_channels(exclude).info
+tmin = limo_epochs['2'].tmin.copy()
+
+
+###############################################################################
+# use epochs metadata to create design matrix for linear regression analyses
+
+# add intercept
+design = limo_epochs['2'].metadata.copy().assign(intercept=1)
+# effect code contrast for categorical variable (i.e., condition a vs. b)
+design['face a - face b'] = np.where(design['face'] == 'A', 1, -1)
+# create design matrix with named predictors
+predictors = ['intercept', 'face a - face b', 'phase-coherence']
+design = design[predictors]
+
+###############################################################################
+# --- run linear regression analysis using scikit-learn ---
+
+# design matrix
+design_matrix = design.copy()
+# name of predictors
+names = predictors.copy()
+
+# data to be analysed
+data = limo_epochs['2'].get_data(picks_eeg)
+# place holder for results
+out = EvokedArray(np.zeros(data.shape[1:]), epochs_info, tmin)
+
+# number of epochs in data set
+n_samples = len(data)
+# channels x time points
+n_features = np.product(data.shape[1:])
+
+# design matrix dimensions
+n_rows, n_predictors = design_matrix.shape
+
+# vectorised (channel) data for linear regression
+y = np.reshape(data, (n_samples, n_features))
```

Comment:
Here we can probably also use the ```mne.decoding.Vectorizer```
Also let's find one language. n_rows or n_samples is the same here.
Using machine learning language would be ok with me. But at least it has to be consistent.
Having `n_features` and `n_predictors` etc is not a good idea.

Reply:
actually `n_targets`

Reply:
Good point. I'm using the vectorizer now, so we dont really need some of this variables anymore. I think I'm sticking with samples and features in the future.